schemas/index.d.ts

```diff
@@ -28,71 +31,279 @@ export type SizingMetricName = "active_hours" | "search_count" | "days_of_use" |
+   */
+  isRollout?: boolean;
   bucketConfig: ExperimentBucketConfig;
-  outcomes?: ExperimentOutcome[] | null;
-  featureIds: string[];
-  branches: (
-    | ExperimentSingleFeatureBranch
-    | ExperimentMultiFeatureDesktopBranch
-    | ExperimentMultiFeatureMobileBranch
-  )[];
+  /**
+   * A list of outcomes relevant to the experiment analysis.
+   */
+  outcomes?: ExperimentOutcome[];
+  /**
+   * A list of featureIds the experiment contains configurations for.
+   */
+  featureIds?: string[];
+  /**
+   * Branch configuration for the experiment.
+   */
+  branches:
+    | ExperimentSingleFeatureBranch[]
+    | ExperimentMultiFeatureDesktopBranch[]
+    | ExperimentMultiFeatureMobileBranch[];
   targeting?: string | null;
-  startDate?: string | null;
+  /**
+   * Actual publish date of the experiment.
+   *
+   * Note that this value is expected to be null in Remote Settings.
+   */
+  startDate: string | null;
+  /**
+   * Actual enrollment end date of the experiment.
+   *
+   * Note that this value is expected to be null in Remote Settings.
+   */
   enrollmentEndDate?: string | null;
-  endDate?: string | null;
-  proposedDuration?: number | null;
-  proposedEnrollment?: number | null;
-  referenceBranch?: string | null;
-  featureValidationOptOut?: boolean | null;
+  /**
+   * Actual end date of this experiment.
+   *
+   * Note that this field is expected to be null in Remote Settings.
+   */
+  endDate: string | null;
+  /**
+   * Duration of the experiment from the start date in days.
+   *
+   * Note that this property is only used during the analysis phase (i.e., not by the SDK).
+   */
+  proposedDuration?: number;
+  /**
+   * This represents the number of days that we expect to enroll new users.
+   *
+   * Note that this property is only used during the analysis phase (i.e., not by the SDK).
+   */
+  proposedEnrollment: number;
+  /**
+   * The slug of the reference branch (i.e., the branch we consider "control").
+   */
+  referenceBranch: string | null;
+  /**
+   * Opt out of feature schema validation. Only supported on desktop.
+   */
+  featureValidationOptOut?: boolean;
+  /**
+   * Per-locale localization substitutions.
+   *
+   * The top level key is the locale (e.g., "en-US" or "fr"). Each entry is a mapping of string IDs to their localized equivalents.
+   *
+   * Only supported on desktop.
+   */
   localizations?: {
-    [k: string]: unknown;
+    [k: string]: {
+      [k: string]: string;
+    };
   } | null;
+  /**
+   * The list of locale codes (e.g., "en-US" or "fr") that this experiment is targeting.
+   *
+   * If null, all locales are targeted.
+   */
   locales?: string[] | null;
+  /**
+   * The date that this experiment was first published to Remote Settings.
+   *
+   * If null, it has not yet been published.
+   */
+  publishedDate?: string;
 }
+/**
+ * Bucketing configuration.
+ */
 export interface ExperimentBucketConfig {
   randomizationUnit: RandomizationUnit;
+  /**
```

Comment:
The old schema had randomizationUnit: string https://github.com/mozilla/nimbus-shared/blob/main/types/experiments.ts

That means new older clients would accept things with invalid randomizationUnits up to the point of bucketing

New clients would reject them at recipe validation time.

Before we publish these as schemas for desktop, do we want to change this @jaredlockhart ?

Reply:
Like as an enum with an exact set of keys?

Reply:
Oh I see, old schema it was just string, new schema it is an exact set of constants.  In theory I would've said not to introduce that as a change ie have the new schema mirror as closely as possible the behaviour of the existing schema, and then handle that change separately.  

Here it's already part of the new schema, even though it's not in use or hooked up to anything yet.

It should be fine, we know what the expected values are, we can see them hard coded in MC.  So it should be fine.

Reply:
Yes the difference is 
```diff
-randomizationUnit: string;
+randomizationUnit: "normandy_id" | "nimbus_id" | "user_id" | "group_id";
```

Reply:
(this will slightly change the behaviour of desktop clients as to when we would reject something with an unsupported randomization unit)